src/features/courses/course-route.ts

```diff
@@ -307,6 +309,78 @@ router.get('/topic/:topicId/version/:userId',
     })
 );
 
+router.put('/topic/:topicId/endExport', 
+    // this call is expected from a microservice, so doesn't go through authentication
+    validate(endBulkExportValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<any, unknown, EndBulkExportRequest.body, EndBulkExportRequest.query>, _res: Response, next: NextFunction) => {
+        const topic = await CourseTopicContent.findOne({
+            where: {
+                id: req.params.topicId,
+                active: true,
+            },
+        });
+
+        if (_.isNil(topic)) {
+            throw new NotFoundError('A endExport PUT was received for a non-existant topic. Was it deleted?');
+        }
+
+        if (_.isNil(req.body.exportUrl)) {
+            topic.exportUrl = null;
+            topic.lastExported = null;
+        } else {
+            topic.exportUrl = req.body.exportUrl;
+        }
+        await topic.save();
+
+        next(httpResponse.Ok('Got it!'));
+    })
+);
+
+router.post('/topic/:topicId/startExport',
+    authenticationMiddleware,
+    validate(bulkExportValidation),
+    // This is due to a typescript issue where the type mismatches extractMap
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<any, unknown, BulkExportRequest.body, BulkExportRequest.query>, _res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+        
+        const params: BulkExportRequest.params = req.params;
+        const query: BulkExportRequest.query = req.query;
+        const professor = await req.session.getUser();
+
+        const topic = await CourseTopicContent.findOne({
+            where: {
+                id: params.topicId,
+                active: true,
+            },
+        });
+
+        if (_.isNil(topic)) {
+            throw new NotFoundError('No such topic was found.');
+        }
+
+        // Since this is a job, we'll always have this by here.
+        const exportDetails = {lastExported: topic.lastExported, exportUrl: topic.exportUrl};
+
+        const helper = new ExportPDFHelper();
+
+        // If we're just checking, return what's already here.
+        if (query.force === false) {
```

Comment:
NIT: This feels like it should be multiple routes, maybe with a common middleware